pkg/backup/remap_crd_version_action.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2020 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package backup
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	v1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+)
+
+// RemapCRDVersionAction inspects CustomResourceDefinition and decides if it is a v1
+// CRD that needs to be backed up as v1beta1.
+type RemapCRDVersionAction struct {
+	logger logrus.FieldLogger
+}
+
+// NewRemapCRDVersionAction instantiates a new RemapCRDVersionAction plugin.
+func NewRemapCRDVersionAction(logger logrus.FieldLogger) *RemapCRDVersionAction {
+	return &RemapCRDVersionAction{logger: logger}
+}
+
+// AppliesTo selects the resources the plugin should run against. In this case, CustomResourceDefinitions.
+func (a *RemapCRDVersionAction) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"customresourcedefinition.apiextensions.k8s.io"},
+	}, nil
+}
+
+// Execute executes logic necessary to check a CustomResourceDefinition and inspect it for characteristics that necessitate saving it as v1beta1 instead of v1.
+func (a *RemapCRDVersionAction) Execute(item runtime.Unstructured, backup *v1.Backup) (runtime.Unstructured, []velero.ResourceIdentifier, error) {
+	a.logger.Info("Executing RemapCRDVersionAction")
+
+	// This plugin is only relevant for CRDs retrieved from the v1 endpoint that were installed via the v1beta1
+	// endpoint, so we can exit immediately if the resource in question isn't v1.
+	apiVersion, ok, err := unstructured.NestedString(item.UnstructuredContent(), "apiVersion")
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "unable to read apiVersion from CRD")
+	}
+	if ok && apiVersion != "apiextensions.k8s.io/v1" {
+		a.logger.Info("Exiting RemapCRDVersionAction, CRD is not v1")
+		return item, nil, nil
+	}
+
+	a.logger.Infof("Unstructured looked like: %#v", item.UnstructuredContent())
```

Comment:
should we bump this down to debug?

Reply:
Actually we can just remove this, it was from writing some tests.